ENH: Segment Features Filters are more consistent with memory allocations

imikejackson opened the pull request:
- All will only allocate a single tuple for feature attribute matrix during preflight
- Feature Attribute Matrix is now resized after the segmentation is completed instead of every iteration
- Code is much more consistent in variable naming and code layout between the 3 filters

imikejackson committed:
ENH: Segment Features Filters are more consistent with memory allocations

- All will only allocate a single tuple for feature attribute matrix during preflight
- Feature Attribute Matrix is now resized after the segmentation is completed instead of every iteration
- Code is much more consistent in variable naming and code layout between the 3 filters

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

nyoungbq commented on src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/EBSDSegmentFeatures.cpp:
```diff
@@ -94,12 +103,12 @@ int64_t EBSDSegmentFeatures::getSeed(int32 gnum, int64 nextSeed) const
   {
-    auto& activeArray = m_DataStructure.getDataAs<UInt8Array>(m_InputValues->activeArrayPath)->getDataStoreRef();
-    auto& cellFeatureAM = m_DataStructure.getDataRefAs<AttributeMatrix>(m_InputValues->cellFeatureAttributeMatrixPath);
+    //    auto& activeArray = m_DataStructure.getDataAs<UInt8Array>(m_InputValues->activeArrayPath)->getDataStoreRef();
+    //    auto& cellFeatureAM = m_DataStructure.getDataRefAs<AttributeMatrix>(m_InputValues->cellFeatureAttributeMatrixPath);
```
```suggestion
```
remove commented out code

nyoungbq commented on src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/EBSDSegmentFeatures.cpp:
```diff
@@ -94,12 +103,12 @@ int64_t EBSDSegmentFeatures::getSeed(int32 gnum, int64 nextSeed) const
-    std::vector<usize> tDims = {static_cast<usize>(gnum) + 1};
-    cellFeatureAM.resizeTuples(tDims); // This will resize the actives array
-    activeArray[gnum] = 1;
+    //    std::vector<usize> tDims = {static_cast<usize>(gnum) + 1};
+    //    cellFeatureAM.resizeTuples(tDims); // This will resize the actives array
+    //    activeArray[gnum] = 1;
```
```suggestion
```
remove commented out code

nyoungbq commented on src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ScalarSegmentFeatures.cpp:
```diff
@@ -269,11 +271,11 @@ int64_t ScalarSegmentFeatures::getSeed(int32 gnum, int64 nextSeed) const
   if(seed >= 0)
   {
-    auto& activeArray = m_DataStructure.getDataRefAs<UInt8Array>(m_InputValues->pActiveArrayPath);
+    // auto& activeArray = m_DataStructure.getDataRefAs<UInt8Array>(m_InputValues->pActiveArrayPath);
```
```suggestion
```

nyoungbq commented on src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ScalarSegmentFeatures.cpp:
```diff
@@ -269,11 +271,11 @@ int64_t ScalarSegmentFeatures::getSeed(int32 gnum, int64 nextSeed) const
-    std::vector<usize> tDims = {static_cast<usize>(gnum) + 1};
-    auto& cellFeaturesAM = m_DataStructure.getDataRefAs<AttributeMatrix>(m_InputValues->pCellFeaturesPath);
-    cellFeaturesAM.resizeTuples(tDims); // This will resize the active array
+    //  std::vector<usize> tDims = {static_cast<usize>(gnum) + 1};
+    //  auto& cellFeaturesAM = m_DataStructure.getDataRefAs<AttributeMatrix>(m_InputValues->pCellFeaturesPath);
+    //  cellFeaturesAM.resizeTuples(tDims); // This will resize the active array
```
```suggestion
```

imikejackson committed:
Update src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/EBSDSegmentFeatures.cpp

Co-authored-by: Nathan Young <nathan.young@bluequartz.net>

imikejackson committed:
Update src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/EBSDSegmentFeatures.cpp

Co-authored-by: Nathan Young <nathan.young@bluequartz.net>

imikejackson committed:
Update src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ScalarSegmentFeatures.cpp

Co-authored-by: Nathan Young <nathan.young@bluequartz.net>

imikejackson committed:
Update src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ScalarSegmentFeatures.cpp

Co-authored-by: Nathan Young <nathan.young@bluequartz.net>

imikejackson committed:
- Unit test: Check for a specific number of features that were created.
- Fix bugs related to reporting the total number of features that were created.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Update error codes that are all the same. -5551.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
- Fix unit tests for ScalarSegmentFeatures

The output has been looked at the committer and found to be correct.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Fix failing unit tests

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Update docs for C++20 mentions and update a pipeline.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>
